src/openassetio-core/tests/hostApi/ManagerTest.cpp

```diff
@@ -413,3 +673,107 @@ TEMPLATE_TEST_CASE(
     }
   }
 }
+
+TEMPLATE_TEST_CASE(
+    "BatchElementError conversion to exceptions when registering", "",
+    (BatchElementErrorMapping<openassetio::UnknownBatchElementException, ErrorCode::kUnknown>),
+    (BatchElementErrorMapping<openassetio::InvalidEntityReferenceBatchElementException,
+                              ErrorCode::kInvalidEntityReference>),
+    (BatchElementErrorMapping<openassetio::MalformedEntityReferenceBatchElementException,
+                              ErrorCode::kMalformedEntityReference>),
+    (BatchElementErrorMapping<openassetio::EntityAccessErrorBatchElementException,
+                              ErrorCode::kEntityAccessError>),
+    (BatchElementErrorMapping<openassetio::EntityResolutionErrorBatchElementException,
+                              ErrorCode::kEntityResolutionError>)) {
+  namespace managerApi = openassetio::managerApi;
+  namespace hostApi = openassetio::hostApi;
+  using trompeloeil::_;
+
+  using ExpectedExceptionType = typename TestType::ExceptionType;
+  static constexpr ErrorCode kExpectedErrorCode = TestType::kErrorCode;
+
+  GIVEN("a configured Manager instance") {
+    const std::shared_ptr<managerApi::ManagerInterface> managerInterface =
+        std::make_shared<openassetio::MockManagerInterface>();
+
+    auto mockManagerInterfacePtr =
+        std::static_pointer_cast<openassetio::MockManagerInterface>(managerInterface);
+
+    // Create a HostSession with our mock HostInterface
+    const managerApi::HostSessionPtr hostSessionPtr = managerApi::HostSession::make(
+        managerApi::Host::make(std::make_shared<openassetio::MockHostInterface>()),
+        std::make_shared<openassetio::MockLoggerInterface>());
+
+    // Create the Manager under test.
+    const hostApi::ManagerPtr manager =
+        hostApi::Manager::make(mockManagerInterfacePtr, hostSessionPtr);
+
+    const openassetio::trait::TraitSet traits = {"fakeTrait", "secondFakeTrait"};
+    auto context = openassetio::Context::make();
+
+    AND_GIVEN(
+        "manager plugin will encounter an entity-specific error when next registering a "
+        "reference") {
+      const openassetio::EntityReference ref = openassetio::EntityReference{"testReference"};
+      const openassetio::EntityReferences refs = {ref};
+
+      auto singleTraitsData = openassetio::TraitsData::make(traits);
+      const openassetio::trait::TraitsDatas singleTraitsDatas = {singleTraitsData};
+
+      const openassetio::BatchElementError expectedError{kExpectedErrorCode, "Some error message"};
+
+      // With success callback side effect
```

Comment:
[fix]
```suggestion
      // With error callback side effect
```

Reply:
Fixed as part of : https://github.com/OpenAssetIO/OpenAssetIO/pull/879/commits/b17da05a778f733c17f627edce68b13b4a4a8057